### PR TITLE
ADBDEV-6257: Fix pg_dump incorrect cleanup for external tables

### PR DESCRIPTION
**Pull request by silent-observer**
Fix pg_dump incorrect cleanup for external tables

Fixes issue #1025. DROP EXTERNAL TABLE in pg_dump output had incorrect syntax,
this patch fixes the typo.

Co-authored-by: eshkinkot \<eshkinkot@gmail.com\>

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/79870

**Comment by bandetto** [comment]
Small note: please escape `<` symbols around "Co-authored-by:" clause with backslashes to avoid them being eaten up by markdown in PR description.

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/79931

File: src/test/regress/input/external_table.source [hunk 1]
```diff
@@ -303,6 +303,13 @@ DROP OWNED BY test_role_issue_12748;
 DROP ROLE test_role_issue_12748;
 DROP PROTOCOL dummy_protocol_issue_12748;
 
+-- Test pg_dump's DROP EXTERNAL TABLE
+-- GitHub Issue #1025: https://github.com/arenadata/gpdb/issues/1025
+CREATE EXTERNAL TABLE issue_1025 (a int) location ('file://@hostname@@abs_srcdir@/data/no/such/place/badt1.tbl' ) format 'csv' (DELIMITER '|');
```
Comment (whitehawk) [inline src/test/regress/input/external_table.source]: Mixed style in the test - a part of the query is in the lower case and the other part is in the upper case. I know that the code right above is written in the similar manner. But the code below is all lower-case.

So, let's do not extend the bad tradition. Better to make it in all-lower-case style.

Comment (hilltracer) [inline src/test/regress/input/external_table.source]: Maybe better to set the schema name during table creation? As the schema name is used in subsequent command and in output file:
```sql
create external table public.issue_1025 (a int) ...
```

**Review by hilltracer — CHANGES_REQUESTED** [review]
It looks fine. I tested the patch with the following commands:

- Create a CSV file:
```sh
echo "1|101" > ~/gpdata/tmp/test_file.csv
echo "2|102" >> ~/gpdata/tmp/test_file.csv
echo "3|103" >> ~/gpdata/tmp/test_file.csv
```

- Create the external table:
```sql
CREATE EXTERNAL TABLE public.ext_test_table
	( id integer, data integer) 
	LOCATION ('file://den-laptop/home/den/gpdata/tmp/test_file.csv')
	FORMAT 'csv' (DELIMITER '|');

SELECT * FROM ext_test_table;
```

- Dump the external table:
```sh
pg_dump -s -c -t public.ext_test_table postgres
```

**Review by whitehawk — APPROVED** [review]

**Review by hilltracer — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/80041

**Pull request merged**